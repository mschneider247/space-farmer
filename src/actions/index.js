export const setUser = user => ({
  type: 'SET_USER_NAME',
  user
});

export const setProposal = proposal => ({
  type: 'SET_PROPOSAL',
  proposal
})

export const addProposal = proposal => ({
  type: 'ADD_PROPOSAL',
  proposal
})