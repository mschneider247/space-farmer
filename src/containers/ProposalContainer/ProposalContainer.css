.space-background {
  background-image: url('../../images/spacefarm.jpg');
  background-repeat: no-repeat;
  background-size: 100%;
  min-height: 600px;
}

.proposal-container {
  position: fixed;
  top: 20%;
  left: 10%;
  display: flex;
}

.proposal-card {
  margin: 1%;
}

h3 {
  background-color: transparent;
}

.section_create {
  text-align: center;
}

.create_btn {
  margin-top: 5%;
}