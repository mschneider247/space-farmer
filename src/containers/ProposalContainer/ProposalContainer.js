import React, { Component } from 'react';
import { Redirect } from 'react-router-dom'
import './ProposalContainer.css';
import { connect } from 'react-redux'
import { addProposal } from '../../actions/index.js'

export class ProposalContainer extends Component {
  constructor() {
    super();
    this.state = {
      newProposal: false,
    }
  }

  handleClick = (event) => {
    event.preventDefault();
    this.setState({newProposal: true})
    let proposal = {
      key: this.props.proposals.length + 1,
      id: this.props.proposals.length + 1,
    }
    this.props.addProposal(proposal)
  }

  checkProposalCreation = () => {
    if (this.state.newProposal === true) {
      return ( 
        <Redirect to='/supplies'></Redirect>
      )
    }
  }

  render() {
    const proposals = this.props.proposals.map(proposal => {
      return (
        <article className="proposal-card">
          <h3>Proposal# {proposal.id}</h3>
          <h3>Final Cost $$ {proposal.finalCost}</h3>
          <h3 className="red-text">PROPOSAL REJECTED</h3>
        </article>
      )
    })

    return (
      <section>
      {console.log("proposals", proposals)}
      {this.checkProposalCreation()}
        <section className="space-background">
          <section className="proposal-container">
            {proposals.length === 0 ? <h3>Start a new proposal below!</h3> : proposals}
          </section>
        </section>
        <section className="section_create">
          <button onClick={(event) => this.handleClick(event)} className="create_btn">Create Proposal</button>
        </section>
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  proposals: state.proposals
})

export const mapDispatchToProps = dispatch => ({
  addProposal: proposal => dispatch(addProposal(proposal))
})

export default connect(mapStateToProps, mapDispatchToProps)(ProposalContainer);