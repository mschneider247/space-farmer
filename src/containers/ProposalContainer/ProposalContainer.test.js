import React from 'react';
import { shallow } from 'enzyme';
import { ProposalContainer, mapDispatchToProps, mapStateToProps } from './ProposalContainer';
import { addProposal } from '../../actions';


describe('Proposal Container', () => {
  it('Proposal Container should match its snapshot', () => {
    const proposals = [{key: 1, id: 1}, {key: 1, id: 2}]
    const wrapper = shallow(<ProposalContainer proposals={proposals}/>)
    expect(wrapper).toMatchSnapshot();
  });
});

describe('ProposalContainer mapDispatchToProps', () => {
  it('Should dispatch proposal when addProposal is called', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = addProposal({ id: 1, key: 1 });
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.addProposal({ id: 1, key: 1 });
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});

describe('Nav mapStateToProps', () => {
  it('Should access the proposals key in the Redux store', () => {
    const mockState = {
      proposals: [{}],
    };
    const expected = { proposals: [{}] };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
