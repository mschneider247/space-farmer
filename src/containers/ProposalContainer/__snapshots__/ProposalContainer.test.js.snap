// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Proposal Container Proposal Container should match its snapshot 1`] = `ShallowWrapper {}`;
